lib/address.js

```diff
@@ -112,6 +112,8 @@ Address.prototype._classifyArguments = function(data, network, type) {
     return Address._transformScript(data, network);
   } else if (typeof(data) === 'string') {
     return Address._transformString(data, network, type);
+  } else if (_.isObject(data) && data.network && data.hash && data.type) {
```

Comment:
Rather than checking for the properties here, could we use preconditions to check that they are set?